clang/test/dpct/use-pure-sycl-queue/main.cu

```diff
@@ -0,0 +1,116 @@
+// UNSUPPORTED: system-windows
+
+// RUN: cp %S/* .
+// RUN: dpct --use-pure-sycl-queue -p=%S -out-root %T --cuda-include-path="%cuda-path/include"
+// RUN: FileCheck --input-file %T/kernel1.dp.cpp --match-full-lines %S/kernel1.cu
+// RUN: FileCheck --input-file %T/kernel2.dp.cpp --match-full-lines %S/kernel2.cu
+// RUN: FileCheck --input-file %T/main.dp.cpp --match-full-lines %S/main.cu
+
+// RUN: grep -x "extern sycl::device dev_ct1;" %T/*.cpp | wc -l > %T/extern_count1.txt
+// RUN: FileCheck --input-file %T/extern_count1.txt --match-full-lines extern_count.txt
+// RUN: grep -x "extern sycl::queue q_ct1;" %T/*.cpp | wc -l > %T/extern_count2.txt
+// RUN: FileCheck --input-file %T/extern_count2.txt --match-full-lines extern_count.txt
+// RUN: grep -x "sycl::device dev_ct1;" %T/*.cpp | wc -l > %T/define_count1.txt
+// RUN: FileCheck --input-file %T/define_count1.txt --match-full-lines define_count.txt
+// RUN: grep -x "sycl::queue q_ct1(dev_ct1, sycl::property_list{sycl::property::queue::in_order()});" %T/*.cpp | wc -l > %T/define_count2.txt
+// RUN: FileCheck --input-file %T/define_count2.txt --match-full-lines define_count.txt
+
+#include "common.cuh"
+#include <stdio.h>
+
+void f() {
+  CUcontext ctx;
+  CUdevice device;
+  int major;
+  int minor;
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1026:{{[0-9]+}}: The call to cuCtxCreate_v2 was removed because it is in --use-pure-sycl-queue mode.
+  // CHECK-NEXT: */
+  // CHECK-NEXT: ctx = device;
+  cuCtxCreate_v2(&ctx, 0, device);
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1026:{{[0-9]+}}: The call to cuDevicePrimaryCtxRetain was removed because it is in --use-pure-sycl-queue mode.
+  // CHECK-NEXT: */
+  // CHECK-NEXT: ctx = device;
+  cuDevicePrimaryCtxRetain(&ctx, device);
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1026:{{[0-9]+}}: The call to cuCtxGetDevice was removed because it is in --use-pure-sycl-queue mode.
+  // CHECK-NEXT: */
+  // CHECK-NEXT: device = 0;
+  cuCtxGetDevice(&device);
+  // CHECK: device = dpct::get_major_version(dev_ct1);
+  cudaDriverGetVersion(&device);
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1026:{{[0-9]+}}: The call to cudaSetDevice was removed because it is in --use-pure-sycl-queue mode.
+  // CHECK-NEXT: */
+  // CHECK-NEXT: 0;
+  cudaSetDevice(device);
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1026:{{[0-9]+}}: The call to cudaGetDevice was removed because it is in --use-pure-sycl-queue mode.
+  // CHECK-NEXT: */
+  // CHECK-NEXT: device = 0;
+  cudaGetDevice(&device);
+  // CHECK: major = dpct::get_major_version(dev_ct1);
+  // CHECK-NEXT: minor = dpct::get_minor_version(dev_ct1);
+  cuDeviceComputeCapability(&major, &minor, device);
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1026:{{[0-9]+}}: The call to cuCtxSetCurrent was removed because it is in --use-pure-sycl-queue mode.
```

Comment:
was removed because it is redundant if it is migrated with option --use-pure-sycl-queue.

Reply:
fixed